buildkite/client/pipeline.go

```diff
@@ -23,8 +24,23 @@ type Pipeline struct {
 	BranchConfiguration string                 `json:"branch_configuration"`
 	Provider            BuildkiteProvider      `json:"provider,omitempty"`
 	ProviderSettings    map[string]interface{} `json:"provider_settings,omitempty"`
-	TeamUUIDs           []string               `json:"team_uuids,omitempty"`
-	Steps               []Step                 `json:"steps,omitempty"`
+	// Buildkite doesn't allow you to create a pipeline if you not an admin or if you a member of more that one team or
+	// none of them. So you unable to create a pipeline and attach the "buildkite_team_pipeline" resource to it after it was
+	// created in this case.
+	//
+	// Note that the update of the field in terraform definition is restricted. Use "buildkite_team_pipeline" to update
+	// teams access level Effectively, it's an initial set of teams that own a pipeline. Once pipeline created, it's not
+	// available for editing.
+	//
+	// Example of workflow could be:
+	// 1. Create a pipeline with teams-owners.
+	// 2. Create a set of "buildkite_team_pipeline" resources to reflect exactly this team set, with
+	//    'access_level = "MANAGE_BUILD_AND_READ"' to reflect existing configuration. Apply.
+	// 3. Update the set of "buildkite_team_pipeline" resources as you need and apply it by targeting
+	//    this resources only, i.e. don't attempt to update the pipeline itself.
+	// 4. Reflect the updated team set in this field in terraform definition.
```

Comment:
This comment explains the Terraform workflow needed to succesfully create a Buildkite pipeline given the current limitations / particularities of the `Buildkite` provider.

WDYT about placing this comment in `canva/infrastructure/buildkite/README.md` instead? In other words, it feels to me that the comment is more valuable for users of the provider than to its code maintainers 😄

Reply:
For 100% we would need a similar description in the `Canva/infrastructure/buildkite/README.md`. The reason why I left it here is just to save as much context on why this change was made in the first place. Otherwise, this will be very difficult to understand why this change is presented. Do you want me to replicate it to `Canva/infrastructure/buildkite/README.md` or just move? 🙂

Reply:
I've split this text sheet into two part, the second will go to the `Canva/infrastructure/buildkite/README.md` 👍